src/httpserver/http_request.hpp

```diff
@@ -135,6 +135,21 @@ class http_request {
      **/
      const std::map<std::string, std::string, http::arg_comparator> get_args() const;
 
+     /**
+      * Method to get or create a file info struct in the map if the provided filename is already in the map, return the exiting file info struct, otherwise create one in the map and return it.
+      * @param upload_file_name he file name the user uploaded (this is the identifier for the map entry)
+      * @result a file info struct file_info_s
+     **/
+     file_info_s &get_or_create_file_info(const char *upload_file_name);
```

Comment:
I think that the output can be marked as const. The input can use a reference to `std::string`